src/lib/OpenEXR/ImfInputFile.h

```diff
@@ -15,25 +15,60 @@
 
 #include "ImfForward.h"
 
-#include "ImfGenericInputFile.h"
 #include "ImfThreading.h"
 
+#include "ImfContext.h"
+
+// TODO: remove this once multipart file is cleaned up
+#include "ImfGenericInputFile.h"
+
 OPENEXR_IMF_INTERNAL_NAMESPACE_HEADER_ENTER
 
+class TiledInputFile;
+
+/// \brief Provides generic access to read an image from an EXR file
+///
+/// There are a suite of classes for accessing image data, depending
+/// on the level of complexity your application demands. This one is
+/// perhaps a good starting point. There is a simpler one in \sa
+/// RgbaInputFile, but that has very noted limitations of only being
+/// for a 16-bit half, RGBA image. This class gives access to
+/// arbitrary channels and data type outputs. It also will convert a
+/// tiled image into a "normal" image, and simplify access if you only
+/// want the first part, first image of a file.
+///
+/// If you will be accessing tiles (say for a renderer), working with
+/// multi-part images, or reading deep data there are other classes
+/// which provide API for handling that complexity more efficiently:
+///
+/// InputFile
```

Comment:
```suggestion
```
The comment says this is a list of classes _other_ than `InputFile`, so the list shouldn't include InputFile itself, or rephrase the comment to indicate this is a list of _all_ `Input` classes.